tcping_test.go

```diff
@@ -1,12 +1,99 @@
 package main
 
 import (
+	"net"
+	"net/netip"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
+// createTestStats should be used to create new stats structs.
+// it uses localhost and 12345 to be tested using the testServerListen.
+// it could fail if netip.ParseAddr has failed.
+func createTestStats(t *testing.T) *stats {
+	addr, err := netip.ParseAddr("127.0.0.1")
+	s := stats{
+		printer: &fakePrinter{},
+		ip:      addr,
+		port:    12345,
+		ticker:  time.NewTicker(time.Second),
+	}
+	if err != nil {
+		t.Errorf("ip parse: %v", err)
+	}
+
+	return &s
+}
+
+// testServerListen creates a new listener
+// on port 12345 and automatically starts it.
+//
+// Use t.Cleanup to close it after the test,
+// so that other tests are not affected.
+//
+// It could fail if net.Listen or Accept has failed.
+func testServerListen(t *testing.T) net.Listener {
+	srv, err := net.Listen("tcp", ":12345")
+	if err != nil {
+		t.Errorf("test server: %v", err)
+	}
+
+	go func() {
+		for {
+			c, err := srv.Accept()
+			if err != nil {
+				return
+			}
+
+			c.Close()
+		}
+	}()
+
+	return srv
+}
+
+func TestProbeSuccess(t *testing.T) {
+	stats := createTestStats(t)
+	stats.ticker = time.NewTicker(time.Nanosecond)
+	srv := testServerListen(t)
+	t.Cleanup(func() {
+		if err := srv.Close(); err != nil {
+			t.Errorf("srv close: %v", err)
+		}
+	})
+
+	expectedSuccessfull := 100
```

Comment:
any specific reason to go with 100?

I assume it would make the tests run 200 seconds more now?